### ISSUE TITLE
initState is being called on the Silidable child when it closes or opens

### ISSUE THREAD

**Issue by davidmartos96** [letsar/flutter_slidable#51]
Is this a bug? This behavior is not expected to happen. One example could be a network request inside the slidable's child, where you don't want to call this operation every time the user interacts with the slidable. 

Here is a sample code. 
```
import 'dart:async';

import 'package:flutter/material.dart';
import 'package:flutter/widgets.dart';
import 'package:flutter_slidable/flutter_slidable.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: "Sample",
      home: Scaffold(
        appBar: AppBar(
          title: Text("The child reloads when sliding"),
        ),
        body: Column(
          children: <Widget>[
            Slidable(
              child: StfulChild(),
              delegate: SlidableBehindDelegate(),
              actions: <Widget>[
                Icon(Icons.share),
              ],
              secondaryActions: <Widget>[
                Icon(Icons.add),
              ],
            )
          ],
        ),
      ),
    );
  }
}

class StfulChild extends StatefulWidget {
  _StfulChildState createState() => _StfulChildState();
}

class _StfulChildState extends State<StfulChild> {
  bool loading = true;

  @override
  void initState() {
    super.initState();
    Future.delayed(Duration(seconds: 1)).then((_) {
      if (mounted)
        setState(() {
          loading = false;
        });
    });
  }

  @override
  Widget build(BuildContext context) {
    return Container(
      color: Colors.green,
      height: 100,
      child: Center(
        child: loading ? CircularProgressIndicator() : Text("Loaded"),
      ),
    );
  }
}

```

**Comment by florianhaar** [comment]
Is there any progress for this bug? It is really anoying that initState is called 1-3 times while the item is slided.

**Comment by frankyvij** [comment]
Hello - can this please be prioritised? Its common to have animations in widgets and/or network request to get data. I have tried to find a workaround, but it seems impossible with current implementation.

**Comment by dineshsheoran24** [comment]
Any progress ??

**Comment by letsar** [comment]
The widget is rebuilding because some widgets are removed/added in the tree during the animation. I will see if I can  remove this behavior in my refactoring.

**Comment by husitawi** [comment]
For info,  this behavior also happens with ValueListenableBuilder in the child's subtree. ( Many thanks for the package! )

**Comment by denisbabineau** [comment]
I've found that if you set a GlobalKey on your Slidable's child it fixes (works around?) this issue. My understanding is that it is caused by the child moving within the widget tree, setting a key allows it to reuse its state under a different parent as explained here:

https://www.youtube.com/watch?v=kn0EOS-ZiIc&feature=youtu.be&t=510

**Comment by letsar** [comment]
I think this will be fixed in the upcoming version. Can someone test it with the revival branch?

**Comment by ofoudane** [comment]
Is this problem fixed. After some investigations, I found out that the problem is related to using InheritedWidget and changing the widget hierarchy when closing the slider. One potential solution would be to only use Stateful and Stateless widgets, and propagating the data for all Slidablechilds.

**Comment by letsar** [comment]
Fixed in 1.0
